src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -20,577 +20,563 @@
 using Roslyn.Utilities;
 using LSP = Roslyn.LanguageServer.Protocol;
 
-namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
+
+/// <summary>
+/// Root type for both document and workspace diagnostic pull requests.
+/// </summary>
+/// <typeparam name="TDiagnosticsParams">The LSP input param type</typeparam>
+/// <typeparam name="TReport">The LSP type that is reported via IProgress</typeparam>
+/// <typeparam name="TReturn">The LSP type that is returned on completion of the request.</typeparam>
+internal abstract partial class AbstractPullDiagnosticHandler<TDiagnosticsParams, TReport, TReturn> : ILspServiceRequestHandler<TDiagnosticsParams, TReturn?>
+    where TDiagnosticsParams : IPartialResultParams<TReport>
 {
-    internal abstract class AbstractDocumentPullDiagnosticHandler<TDiagnosticsParams, TReport, TReturn>(
```

Comment:
this had been annoying me for a while.  i'd go to the file for AbstractPullDiagnosticHandler and would see this guy.